### PR TITLE
Fix stopListening when changing a prop of type "state"

### PR DESCRIPTION
**Pull request by janpaul123**
We used to just call `this.stopListening` on a property of type
"state" whenever we changed it out for some other object. However,
this also unbinds any other listeners on that object, unexpectedly.

Instead, we should keep a reference to the handler, which we do by
changing `_getEventBubblingHandler` into
`_getCachedEventBubblingHandler` and saving it for the `propertyName`.
This way, we can unbind just the handler we just set, and leave other
event listeners untouched.

### CODE REVIEW

File: ampersand-state.js [hunk 1]
```diff
@@ -756,11 +760,11 @@ var dataTypes = {
             // if this has changed we want to also handle
             // event propagation
             if (previousVal) {
-                this.stopListening(previousVal);
+                this.stopListening(previousVal, 'all', this._getCachedEventBubblingHandler(attributeName));
```
Comment (hassoncs) [inline ampersand-state.js:763]: Should we remove the cached version from `this._eventBubblingHandlerCache` once it's unbound so it doesn't stay in memory?

**Comment by hassoncs** [comment]
Looks good to me, just one comment. :+1:

File: ampersand-state.js (hunk 1 above)
Comment (janpaul123) [inline ampersand-state.js:763]: No, because a few lines down we use it again. :)

Comment (janpaul123) [inline ampersand-state.js:763]: Also, AFAIK the number of properties is fixed so this should never get out of hand.

**Pull request merged**